pcdsdevices/gasdet.py

```diff
@@ -0,0 +1,345 @@
+        kind='config'
+    )
+
+    def __init__(self,
+                 prefix,
+                 module,
+                 platform=None,
+                 name='acqiris_module',
+                 **kwargs):
+        self.module = module
+        self.prefix = prefix
+        self.platform = platform
+        if len(self.module) != 3:
+            raise ValueError("acqiris 'module'"
+                             + " expects a 3-digit module identifier,"
+                             + " e.g. '240'")
+        self._mod_pref = self.module[:2]
+        super().__init__(prefix, name=name, **kwargs)
+
+    def start(self):
+        """
+        Start the acqiris DAQ
+        """
+        self.run_state.put(0)
+
+    def stop(self):
+        """
+        Stop the acqiris DAQ
+        """
+        self.run_state.put(1)
+
+
+class gasdet(Device):
+    """
+    Class to read out pulse energy from a PMT/diode as digitized
+    by an acqiris in the GasDetDAQ EPICS IOC.
+
+    Parameters
+    ----------
+    prefix : ``str``
+        The PV base of the digitizer.
+    platform : ``str``
+        This a legacy PV segment found in the GasDetDAQ IOC.
+        For example, '202' applies to
+        the HXR gas detector system... for some reason.
+    module : ``str``
+        3-digit number representing the digitizer board, e.g. '240', '360'.
+    channel : ``str``
+        The digitizer channel.
+
+    """
+    mJ = Cpt(
+        EpicsSignalRO,
+        ':ENRC',
+        kind='hinted'
+    )
+    time = Cpt(
+        EpicsSignalRO,
+        ':TIME',
+        kind='hinted'
+    )
+    cal = Cpt(
+        EpicsSignalRO,
+        ':CALI',
+        kind='hinted'
+    )
+    offset = Cpt(
+        EpicsSignalRO,
+        ':OFFS',
+        kind='hinted'
+    )
+    BG_start_index = Cpt(
+        EpicsSignalRO,
+        ':BSTR',
+        kind='hinted'
+    )
+    BG_stop_index = Cpt(
+        EpicsSignalRO,
+        ':BSTP',
+        kind='hinted'
+    )
+    BG_sum = Cpt(
+        EpicsSignalRO,
+        ':BSUM',
+        kind='hinted'
+    )
+    pulse_start_index = Cpt(
+        EpicsSignalRO,
+        ':STRT',
+        kind='hinted'
+    )
+    pulse_stop_index = Cpt(
+        EpicsSignalRO,
+        ':STOP',
+        kind='hinted'
+    )
+    pulse_sum = Cpt(
+        EpicsSignalRO,
+        ':PSUM',
+        kind='hinted'
+    )
+
+    def __init__(self,
+                 prefix,
+                 *,
+                 platform=None,
+                 module=None,
+                 channel=None,
+                 name='gasdet',
+                 **kwargs):
+        self.module = module
+        self.prefix = prefix
+        self.platform = platform
+        if self.module and len(self.module) != 3:
+            raise ValueError("acqiris 'module'"
+                             + " expects a 3-digit module identifier,"
+                             + " e.g. '240'")
+        else:
+            if self.module:
+                self._mod_pref = self.module[:2]
```

Comment:
This seems wrong

Reply:
thanks

Reply:
There's probably a better way to do it but I do not have a suggestion. You're fighting with legacy PVs.

Reply:
Am I missing something? `_mod_pref`, `module`, `platform`, etc. aren't used in this class - there aren't even any FCpts.

Reply:
No, I'm seeing the same thing now. These are used in the channel class which isn't linked here. I got confused reading this file and the two classes merged together in my head.